ci/include-grouping.py

```diff
@@ -0,0 +1,159 @@
+                        includes = []
+
+                    # print skipped lines
+                    for skipped_line in skipped_lines:
+                        print(skipped_line, end="", file=out_file)
+                    skipped_lines = []
+
+                    print(line, end="", file=out_file)
+
+    # move out_file to file
+    os.rename(file + ".tmp", file)
+
+    pass
+
+
+def print_sorted_includes(includes, file=sys.stdout):
+    default_group_key = 1
+    groups = {}
+
+    matches = [
+        {
+            "name": "system includes",
+            "matcher": lambda x: x.startswith("<"),
+            "group_key": -2,
+            "priority": 0,
+        },
+        {
+            "name": "naked postgres includes",
+            "matcher": lambda x: "/" not in x,
+            "group_key": 0,
+            "priority": 9,
+        },
+        {
+            "name": "postgres.h",
+            "list": ['"postgres.h"'],
+            "group_key": -1,
+            "priority": -1,
+        },
+        {
+            "name": "naked citus inlcudes",
+            "list": ['"citus_version.h"', '"pg_version_compat.h"'],
+            "group_key": 3,
+            "priority": 0,
+        },
+        {
+            "name": "positional citus includes",
+            "list": ['"distributed/pg_version_constants.h"'],
+            "group_key": 4,
+            "priority": 0,
+        },
+        {
+            "name": "columnar includes",
+            "matcher": lambda x: x.startswith('"columnar/'),
+            "group_key": 5,
+            "priority": 1,
+        },
+        {
+            "name": "distributed includes",
+            "matcher": lambda x: x.startswith('"distributed/'),
+            "group_key": 6,
+            "priority": 1,
+        },
+    ]
+
+    matches.sort(key=lambda x: x["priority"])
+
+    common_system_include_error_prefixes = ["<nodes/", "<distributed/"]
+
+    for include in includes:
+        # extract the group key from the include
+        include_content = include.split(" ")[1]
+
+        # fix common system includes which are secretly postgres or citus includes
+        for common_prefix in common_system_include_error_prefixes:
+            if include_content.startswith(common_prefix):
+                include_content = '"' + include_content.strip()[1:-1] + '"'
+                include = include.split(" ")[0] + " " + include_content + "\n"
+                break
+
+        group_key = default_group_key
+        for matcher in matches:
+            if "list" in matcher:
+                if include_content.strip() in matcher["list"]:
+                    group_key = matcher["group_key"]
+                    break
+
+            if "matcher" in matcher:
+                if matcher["matcher"](include_content):
+                    group_key = matcher["group_key"]
+                    break
+
+        if group_key not in groups.keys():
+            groups[group_key] = []
+        groups[group_key].append(include)
```

Comment:
if we declare [groups](https://github.com/citusdata/citus/pull/7326/files#diff-4ef8be7a6b340207edfe9e3923061e33b918e3c0cb62dac9ec6989f882ddb716R61) as `collections.defaultdict(set)` this block could be written as:
```suggestion
        groups[group_key].add(include)
```

What will also allow to simplify the printing (see next comment).